### DISCUSSION TITLE
Need help on migrating a HelmRelease from GitRepository source to HelmRepository

### DISCUSSION THREAD

**Discussion by OuesFa** [fluxcd/flux2#2992] [Q&A] [answered]
I'm deploying a `HelmRelease` resource by pointing to the chart using a `GitRepository` resource in the `spec.chart.spec.sourceRef.kind`  
I'm trying to use the upstream chart directly from the HelmRepository provided by the chart maintainers instead of my forked git repository. 
I tried removing the `gitrepo` resource from my kube cluster & adding the `helmrepo` resource & change the `spec.chart.spec.sourceRef` config  in my HelmRelease to point to the new `helmrepo`. 

I'm using the same chart version. 

I was expecting the HelmRelease to be updated correctly, meaning that it should point the helmrepo instead of the gitrepo. 

It doesn't seem to work though, when I describe the HelmRelease I get the following
```
strimzi-kafka-operator   False   HelmChart 'dbre/dbre-strimzi-kafka-operator' is not ready   186d
``` 
And the helm controller pod keeps telling me the same continuously.

Knowing that 'dbre/dbre-strimzi-kafka-operator' is the "old" local gitrepo chart path, the new one is 'strimzi/strimzi-kafka-operator' located in the `https://strimzi.io/charts/` helmrepo    

Note: I'm using `Flux Version: v0.27.1`

Any help on this would be much appreciated 🙏

**Comment by stefanprodan** [comment, marked as answer]
When using Helm repos you can't have a path as the chart name, it must be the actual name e.g. `strimzi-kafka-operator`.

**Reply by OuesFa** [reply]
Thanks a lot @stefanprodan for your quick answer, it works fine with the fix you suggested. 
```
strimzi-kafka-operator   True      Release reconciliation succeeded                            186d
```

I ask another question related to the same chart, the strimzi operator chart contains a `crds` folder, before, with FluxV1 I managed the crds manually (I applied them with kubectl)
From now on I take the opportunity to switch the CRDs management to FluxV2 by putting the following two configs 
`spec.install.crds` & `spec.upgrade.crds` to `CreateReplace`

Knowing that the CRDs are already installed on the kubernetes cluster, is there a risk that FluxV2 doesn't want to take them into account (because they are missing some annotations it expects for instance)? 
Is there also a risk in your opinion that FluxV2 will delete the CRDs? It seems completely unlikely to me but I ask the question because it would mean that the associated CRs would be deleted too which would be compromising?

**Reply by stefanprodan** [reply]
> Is there also a risk in your opinion that FluxV2 will delete the CRDs?

No, Flux will never delete CRDs that are inside a chart in the `crds` dir.

**Reply by OuesFa** [reply]
Ok thank you! 

Something seems strange to me though, I don't see any annotation indicating that fluxv2 handles the CRDs.
When I do this command on anther resource handled by flux 
```
k get hr strimzi-kafka-operator -oyaml |grep -i flux                                                                      
  - finalizers.fluxcd.io
    kustomize.toolkit.fluxcd.io/name: dbre
    kustomize.toolkit.fluxcd.io/namespace: flux-system
```
I can see the annotations indicating that it is managed by flux.

On the other hand, when I do the same on one of the CRDs that is supposed to be managed by flux, because it is present in the `crds` folder of the operator's chart
```
❯ k get crd kafkas.kafka.strimzi.io -oyaml |grep -i flux
Nothing
```

Is this normal?
 
Since helm doesn't handle CRDs, I would expect to see the kustomize annotations of flux that indicate that the resources were indeed applied by flux right?

What am I missing here please?

**Reply by stefanprodan** [reply]
See https://github.com/fluxcd/helm-controller/issues/476

**Reply by OuesFa** [reply]
Oh ok thanks for the pointer 🙏 
Do you think fixing this bug could be a candidate for a first contribution from someone who doesn't know the project at all or is it too complex?

**Reply by stefanprodan** [reply]
We're rewriting helm-controller and this will be fixed as part of the rewrite effort.

**Reply by OuesFa** [reply]
Ok great!
Thank you for your help!
